SEED-platform/seed
# #1614, #1615, #2020, #2114 - DQ refactor

#### Any background context you want to provide?
Added condition check column for null field values filtering.
Currently when multiple rules set up for the same field, properties missing values for this field will be reported as warning or error multiple times.

#### What's this PR do?
1. Removed report redundancy on null values;
2. Applied labels in data quality report:
<img width="978" alt="Screen Shot 2020-05-07 at 12 14 18 PM" src="https://user-images.githubusercontent.com/49968203/81330451-532c6c80-905d-11ea-9cd0-2eb16917575e.png">
3. Removed 'required' and 'not null' columns;
4. Added rule condition check for value existence (required/null), range (min/max), and text (contain/not contain)
5. Removed required/not null checkbox columns and add condition checks for both;
6. Alert users of empty input by outlining text box in red if rule must include/exclude a string;
7. Migration added to update current db. **'Required' and 'not null' fields should not be removed in migration until this PR is approved.**
<img width="1111" alt="Screen Shot 2020-03-20 at 9 45 17 AM" src="https://user-images.githubusercontent.com/49968203/77180786-0a5e2b80-6a90-11ea-84d2-9cc9ef990a54.png">
<img width="1356" alt="Screen Shot 2020-04-30 at 2 30 19 PM" src="https://user-images.githubusercontent.com/49968203/80756357-96cc2700-8aef-11ea-97ef-109dbe688974.png">

#### How should this be manually tested?
Select a condition for check:
1. Required/Not Null checks for null field values and report once if captured. This option disables min/max value and units columns;
3. If condition checks is selected for range, min and/or max sections are enabled for user to fill in with min/max values;
4. If condition check is selected for 'include' or 'not include', rule checks for included/excluded string.

#### What are the relevant tickets?
#1614, #1615, #2020, #2114   

#### Screenshots (if appropriate)
Set up a blank range rule for not null value check for Site EUI:
<img width="1311" alt="Screen Shot 2020-03-04 at 6 42 00 PM" src="https://user-images.githubusercontent.com/49968203/75939619-f1c30400-5e47-11ea-9bc4-47c7ba91a140.png">

Null site eui is reported once for the properties that are missing it along with other properties that have eui but are invalid:
<img width="1060" alt="Screen Shot 2020-03-04 at 6 45 28 PM" src="https://user-images.githubusercontent.com/49968203/75939741-54b49b00-5e48-11ea-88fc-c1fe5cc468d5.png">
